git_wrapper/repo.py

```diff
@@ -92,27 +88,17 @@ def describe(self, sha):
                 ret_data['patch'] = output[1]
         return ret_data
 
-    def add_remote(self, name, url):
-        """Adds a remote to the given repo
-
-            :param str name: The name for the remote
-            :param str url: The url to use for the remote
-            :return bool: True if the remote was added, False otherwise
-        """
-        self.logger.debug("Adding remote %s (%s) to repo %s", name, url, self.repo.working_dir)
-        ret_status = False
-
-        try:
-            remote = self.repo.create_remote(name, url)
-        except git.CommandError as ex:
-            self.logger.debug("Failed to create new remote %s (%s). Error: %s", name, url, ex)
-            return ret_status
+    @property
+    def remote(self):
+        """Return object to act on the repo's remotes"""
+        if not self._remote:
+            self._remote = GitRemote(git_repo=self, logger=self.logger)
+        return self._remote
 
-        try:
-            remote.update()
-            ret_status = True
-        except git.CommandError as ex:
-            self.logger.debug("Failed to update new remote %s (%s), removing it. Error: %s", name, url, ex)
-            self.repo.delete_remote(remote)
+    @remote.setter
+    def remote(self, new_remote):
+        """Set up object to interact with Remotes
 
-        return ret_status
+            :param git_wrapper.remote.GitRemote new_remote: An already constructed GitRemote object to use
+        """
+        self._remote = new_remote
```

Comment:
We should use a type check here to make sure we get a GitRemote instance and not some other object. We could then raise a type error if we get something unexpected.